packages/odyssey-react-mui/src/test-selectors/querySelector.ts

```diff
@@ -11,188 +11,238 @@
+    };
+
+    const selectChild = <
+      FeatureName extends LocalFeatureTestSelector extends FeatureSelector
+        ? keyof LocalFeatureTestSelector["feature"]
+        : keyof FeatureSelector,
+      ChildQueryMethod extends QueryMethod,
+    >(
+      childProps: {
+        featureName: FeatureName;
+      } & InnerQuerySelectorProps<
+        LocalFeatureTestSelector extends FeatureSelector
+          ? LocalFeatureTestSelector["feature"][FeatureName]
+          : FeatureTestSelector,
+        ChildQueryMethod
+      >,
+    ) => {
+      if (!capturedElement) {
+        throw new ElementError(
+          "No child HTML element available",
+          containerElement,
+        );
+      }
+
+      if (!("feature" in featureTestSelector)) {
+        throw new Error("Missing `feature` in `FeatureTestSelector`");
+      }
+
+      type Options = Record<
+        LocalFeatureTestSelector extends FeatureSelector
+          ? LocalFeatureTestSelector["feature"][FeatureName] extends TestSelector
+            ? keyof LocalFeatureTestSelector["feature"][FeatureName]["selector"]["options"]
+            : string
+          : string,
         string | RegExp
-      >
-    : never;
-  testSelectors: TestSelectors;
-}) => {
-  const element =
-    "selector" in testSelectors
-      ? getByQuerySelector({
-          canvas,
-          method: testSelectors.selector.method,
-          options:
-            templateArgsProp && testSelectors.selector.options
-              ? Object.fromEntries(
-                  Object.entries(testSelectors.selector.options).map(
-                    ([key, value]) => [
-                      key,
-                      interpolateString(value, templateArgsProp),
-                    ],
-                  ),
-                )
-              : testSelectors.selector.options,
-          ...(testSelectors.selector.method === "ByRole"
+      >;
+
+      return querySelector(
+        featureTestSelector.feature[
+          childProps.featureName
+        ] as LocalFeatureTestSelector extends FeatureSelector
+          ? LocalFeatureTestSelector["feature"][FeatureName]
+          : FeatureTestSelector,
+      )(
+        // @ts-expect-error: Type '{ role?: AriaRole | undefined; options?: Record<LocalFeatureTestSelector extends FeatureSelector ? LocalFeatureTestSelector["feature"][FeatureName] extends TestSelector ? keyof LocalFeatureTestSelector["feature"][FeatureName]["selector"]["options"] : string : string, string | RegExp> | undefined; element: HTMLElement...' is not assignable to type '(LocalFeatureTestSelector extends FeatureSelector ? LocalFeatureTestSelector["feature"][FeatureName] : FeatureTestSelector) extends { ...; } ? Role extends AriaRole[] ? { ...; } : object : object'.ts(2345)
+        // `as featureTestSelector.feature[featureName]` narrows the props down enough that TypeScript errors here. We're passing the correct information, but it doesn't know that, and it's difficult to fix this. -Kevin Ghadyani
```

Comment:
Would be interested in some more info on what the exact mismatch between the expected types is here. What is the part that doesn't line up? Obviously theres something to do with narrowing the feature[FeatureName] type, but it seems like theres a fair amount of info being omitted from the error message

Reply:
I think it's something to do with `options` specifically. But even if I uncomment these things, it errors.

If you wanna take a look, I'd gladly accept.

Changing this:
```ts
querySelector(
  featureTestSelector.feature[
    childProps.featureName
  ] as LocalFeatureTestSelector extends FeatureSelector
    ? LocalFeatureTestSelector["feature"][FeatureName]
    : FeatureTestSelector,
)
```

Into this, then it works:
```ts
querySelector(
  featureTestSelector.feature[
    childProps.featureName
  ],
)
```

But the types don't show correctly for deeply-nested features. `FeatureName` becomes `string` or `unknown`.

![FeatureName incorrect](image.png)

---

The proper way to handle this is to simply return `querySelector(testSelectors, element)(featureName)`, but then you have to call it like:
```ts
selectChild("listItem")({
  options: {
    label: /Destination/
  }
})
```

I had this before but didn't think the currying made sense to people. The API was super wonky to use. I wish I had an easier way of accepting both a feature name and a set of optional options.